src/client/app/containers/import.ts

```diff
@@ -119,17 +122,31 @@ export class Import {
 
     strings = Strings();
 
-    constructor(private _store: Store<fromRoot.State>) {
+    constructor(
+        private _effects: UIEffects,
+        private _request: Request,
+        private _store: Store<fromRoot.State>
+    ) {
         this._store.dispatch(new Snippet.LoadSnippetsAction());
         this._store.dispatch(new Snippet.LoadTemplatesAction());
 
         this._store.select(fromRoot.getCurrent)
             .do(snippet => this.activeSnippetId = snippet ? snippet.id : null)
             .filter(snippet => snippet == null)
-            .subscribe(() => this._store.dispatch(new UI.ToggleImportAction(true)));
+            .subscribe(() => {
+                if (!this.hasViewUrlSetting()) {
+                    this._store.dispatch(new UI.ToggleImportAction(true));
+                }
+            });
 
         this.showLocalStorageWarning = !(storage.settings.get('disableLocalStorageWarning') as any === true);
         this.showImportWarning = !(storage.settings.get('disableImportWarning') as any === true);
+        this.importViewSnippet()
+            .then(() => {
+                Office.context.document.settings.remove(correlationIdPropertyName);
```

Comment:
It may be nice to leave correlationId in the settings, to be able to track when users open the document more than once. We can differentiate from the initial open due to the fact the view URL is removed.

Reply:
Changed logic to leave this in the settings. Verified that the `saveAsync` API doesn't delete it when saving the file.